From b7616bd28837a58e18bff20c43fd908436e3d23b Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 13 Dec 2022 16:45:51 +0100
Subject: [PATCH 2/5] x86: Compress vmlinux with zstd -19 instead of -22

This gives slightly bigger kernel but it avoids run out of memory on 32
bit kernels with the error:
zstd kernel compression error 11

https://forums.gentoo.org/viewtopic-p-8641020.html#8641020
Signed-off-by: Achill Gilgenast <achill@achill.org>
---
 arch/x86/boot/compressed/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/boot/compressed/Makefile b/arch/x86/boot/compressed/Makefile
index 74657589264d..587671a8c5b0 100644
--- a/arch/x86/boot/compressed/Makefile
+++ b/arch/x86/boot/compressed/Makefile
@@ -144,7 +144,7 @@ $(obj)/vmlinux.bin.lzo: $(vmlinux.bin.all-y) FORCE
 $(obj)/vmlinux.bin.lz4: $(vmlinux.bin.all-y) FORCE
 	$(call if_changed,lz4_with_size)
 $(obj)/vmlinux.bin.zst: $(vmlinux.bin.all-y) FORCE
-	$(call if_changed,zstd22_with_size)
+	$(call if_changed,zstd_with_size)
 
 suffix-$(CONFIG_KERNEL_GZIP)	:= gz
 suffix-$(CONFIG_KERNEL_BZIP2)	:= bz2
-- 
2.52.0

